src/export.rs

```diff
@@ -0,0 +1,235 @@
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::{actions::Action, state::GameOutcome, types::PlayedCard, State};
+
+/// Training data export structures
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct TrainingData {
+    pub games: Vec<GameData>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GameData {
+    pub game_id: String,
+    pub winner: Option<usize>, // 0, 1, or None for tie
+    pub decisions: Vec<Decision>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Decision {
+    pub turn: u8,
+    pub acting_player: usize,
+    pub state: GameState,
+    pub possible_actions: Vec<ActionData>,
+    pub action_chosen: ActionData,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub action_scores: Option<Vec<f64>>, // From expectiminimax if available
+}
+
+/// Simplified game state for training
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GameState {
+    pub turn_count: u8,
+    pub current_player: usize,
+    pub points: [u8; 2],
+    pub hands_size: [usize; 2], // Just hand sizes for privacy
+    pub in_play_pokemon: [[Option<PokemonState>; 4]; 2],
+    pub current_energy: Option<String>, // Energy type as string
+    pub has_played_support: bool,
+    pub has_retreated: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PokemonState {
+    pub card_name: String,
+    pub hp: u32,
+    pub remaining_hp: u32,
+    pub attached_energy: Vec<String>, // Energy types as strings
+    pub attached_tool: Option<String>,
+    pub played_this_turn: bool,
+    pub poisoned: bool,
+    pub paralyzed: bool,
+    pub asleep: bool,
+    pub stage: u8,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ActionData {
+    pub action_type: String,
+    pub description: String,
+    pub actor: usize,
+}
```

Comment:
These three struct look suspiciously similar to the ones in the core. Could we use the ones in the core instead? It'd be easier to maintain going forward if we just have one struct (less code, and it'll be easier for people to keep track of concepts in the repo).

I imagine we also would remove the need for the ::from methods below, making the codebase simpler.